clang/include/clang/Basic/Diagnostic.h

```diff
@@ -1024,6 +1024,9 @@ class DiagnosticsEngine : public RefCountedBase<DiagnosticsEngine> {
   /// @}
 };
 
+/// Declare the friend function for testing to avoid GCC warning.
```

Comment:
couldn't we just put this at the top of the unittest file where this is defined/used?  Just do:

`namespace clang {
<declaration>
}`

It seems silly to do this for a function that is only used in the unittests.